### ISSUE TITLE
Background Usage

### ISSUE THREAD

**Issue by LeviWilliams** [mrousavy/react-native-google-nearby-messages#12]
This is more of a feature request/question. I've been looking into using this library to still detect devices while running in the background. Obviously I could register a headless component for android but this does not work in iOS. We would have to write a native iOS bridge to stay cross-platform and support this running in the background. 

I do not have much experience in this and thus I'm wondering if this is even a possible addition to this library. Or if it's even possible to do some other way? I know native apps use nearby messages in the background normally. Would love to hear your thoughts.

**Comment by mrousavy** [comment]
Yes, this is certainly possible! (as far as the native implementation goes at least)

See the official google nearby docs, they have a part specific for background tasks.

I'm not planning on adding this myself in the near future - it should not be a lot of code though so if you do, create a PR and I'll review & merge it. :)

**Comment by mrousavy** [comment]
@LeviWilliams The Google Nearby Messages documentation has an entry for background operation [here](https://developers.google.com/nearby/messages/ios/pub-sub#background_operation), so you could try to set the background mode in my swift code to true yourself and experiment with it.

If you got something working, let me know, so we can implement this feature.

**Comment by zaptrem** [comment]
+1 on this. Don't currently have an Android dev/testing workflow set up to work on it, but I'll follow this with interest!

**Comment by LeviWilliams** [comment]
I also don't have much of a workflow set up, I've messed around with the Swift a bit but nothing substantial yet. If anyone has more experience dealing with this I'd definitely help test the PR.
